### PR TITLE
Fix hexdump_iter for actual file objects, like those used by phd

### PR DESCRIPTION
**Pull request by zachriggle**
This is a regression introduced somewhere between 3.3.1 and 3.3.3 from earlier this week.

### CODE REVIEW

**Comment by zachriggle** [comment]
@Idolf ping

File: pwnlib/util/fiddling.py [hunk 1]
```diff
@@ -615,14 +616,15 @@ def hexdump_iter(fd, width=16, skip=True, hexii=False, begin=0, style=None,
...
         # Save the current file offset
-        cur = fd.seek(0, os.SEEK_CUR)
+        cur = fd.tell()
 
         # Determine the total size of the file
         fd.seek(0, os.SEEK_END)
-        total = fd.tell()
+        total = fd.tell() - cur
+
+        # Restore the file offset, and
```
Comment (TethysSvensson) [inline pwnlib/util/fiddling.py:625]: And what?

Comment (zachriggle) [reply pwnlib/util/fiddling.py:625]: No "and", just a dangling lost thought in a comment.

This commit is cherry-picked from `dev` (f46112e8), which has the same dangling comment -- but also has tests which pass.

**Comment by TethysSvensson** [comment]
`phd` is still broken, even with this commit.

```sh
head -c 128 /bin/sh | phd
```

**Pull request merged**